wgpu/gui/qt.py

```diff
@@ -198,6 +260,175 @@ def close(self):
     def is_closed(self):
         return not self.isVisible()
 
+    # Auto event API
+
+    def _emit_event(self, event):
+        try:
+            self.handle_event(event)
+        except Exception:
+            # Print exception and store exc info for postmortem debugging
+            exc_info = list(sys.exc_info())
+            exc_info[2] = exc_info[2].tb_next  # skip *this* function
+            sys.last_type, sys.last_value, sys.last_traceback = exc_info
+            traceback.print_exception(*exc_info)
+
+    def handle_event(self, event):
+        """Handle an incoming event.
+
+        Subclasses can overload this method. Events include widget
+        resize, mouse/touch interaction, key events, and more. An event
+        is a dict with at least the key event_type. For details, see
+        https://jupyter-rfb.readthedocs.io/en/latest/events.html
+        """
+        event_type = event.get("event_type")
+        for callback in self._event_handlers[event_type]:
+            callback(event)
+
+    def add_event_handler(self, *args):
+        """Register an event handler.
+
```

Comment:
We're tri-plicating this code now, let's put it in a base class ... in a next PR.

Reply:
I've made #230 to track this!